test/externalTests/external_test.py

```diff
@@ -0,0 +1,368 @@
+from argparse import ArgumentParser
+
+import json
+from string import Template
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, Tuple
+
+import re
+from abc import ABCMeta, abstractmethod
+
+SOLC_FULL_VERSION_REGEX = re.compile(r'^[a-zA-Z: ]*(.*)$')
+SOLC_SHORT_VERSION_REGEX = re.compile(r'^([0-9.]+).*\+|\-$')
+
+CURRENT_EVM_VERSION: str = "london"
+AVAILABLE_PRESETS: Tuple[str] = (
+    "legacy-no-optimize",
+    "ir-no-optimize",
+    "legacy-optimize-evm-only",
+    "ir-optimize-evm-only",
+    "legacy-optimize-evm+yul",
+    "ir-optimize-evm+yul"
+)
+
+# TODO: add extra compiler settings and set default values
+compiler_input_tmpl = Template("""
+{
+    "optimizer": {
+        "enabled": ${optimizer},
+        "details": {
+            "yul": ${yul}
+        }
+    },
+    "evmVersion": \"${evm_version}\",
+    "viaIR": ${via_ir}
+}""")
+
+
+def settings_from_preset(preset, evm_version) -> Dict:
+    if preset not in AVAILABLE_PRESETS:
+        raise RuntimeError(
+            f"Preset \"{preset}\" not found. Please select one or more of the available presets:\n{' '.join(map(str, AVAILABLE_PRESETS))}")
+    switch = {
+        "legacy-no-optimize": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="false",
+            optimizer="false",
+            yul="false"
+        ),
+        "ir-no-optimize": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="true",
+            optimizer="false",
+            yul="false"
+        ),
+        "legacy-optimize-evm-only": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="false",
+            optimizer="true",
+            yul="false"
+        ),
+        "ir-optimize-evm-only": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="true",
+            optimizer="true",
+            yul="false"
+        ),
+        "legacy-optimize-evm+yul": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="false",
+            optimizer="true",
+            yul="true"
+        ),
+        "ir-optimize-evm+yul": compiler_input_tmpl.substitute(
+            evm_version=evm_version,
+            via_ir="true",
+            optimizer="true",
+            yul="true"
+        )
+    }
+    return json.loads(switch.get(preset))
+
+
+def run_cmd(command: str, env: dict = None, logfile: str = None) -> int:
+    """
+    Args:
+        command: command to run
+        logfile: log file name
+        env:     dictionary holding key-value pairs for bash environment variables
+
+    Returns:
+        int: The exit status of the command. Exit status codes are:
+            0     -> Success
+            1-255 -> Failure
+    """
+    if not logfile:
+        logfile = os.devnull
+    if not env:
+        env = os.environ.copy()
```

Comment:
The way it is now, it will treat an empty dict and an empty string as if the options were not specified at all. So you can't e.g. explicitly run it with empty env.

```suggestion
    if logfile is None:
        logfile = os.devnull
    if env is None:
        env = os.environ.copy()
```

In general I think it's better to compare explicitly when the variable is not a boolean. Otherwise it's easy to forget that `0`, `""`, `{}`, `[]`, etc. are also false. Especially `0` is tricky and a source of bugs.